oak/server/rust/oak_runtime/src/channel.rs

```diff
@@ -29,6 +29,8 @@ use crate::platform;
 use crate::{Message, RuntimeRef};
 
 type Messages = VecDeque<Message>;
+type WaitingThreads =
+    platform::Mutex<HashMap<platform::thread::ThreadId, Weak<platform::thread::Thread>>>;
```

Comment:
The `ThreadId` never gets used (and besides could be recovered from `Thread::id()`) so why not just make this `Mutex<Vec<Thread>>` ?

Reply:
Done